foundationdb/tests/watch.rs

```diff
@@ -71,8 +71,10 @@ fn test_watch_without_commit() {
         .and_then(|trx| {
             eprintln!("setting watch");
 
-            // trx will be dropped without `commit`, so a watch will be canceled
-            trx.watch(KEY)
+            let f = trx.watch(KEY);
+            // `trx` is canceled, so a watch will be canceled
+            trx.cancel();
```

Comment:
transactions are still implicitly canceled with drop, right?

Reply:
The API behavior is changed as you noticed. Now transaction future is handled by `TrxFuture` not `FdbFuture`, and `TrxFuture` maintains reference count for `Transaction` because pointer to the transaction is needed to check backoff behavior on error (more specifically, `fdb_transaction_on_error` requires `*mut fdb::FDBTransaction`).

A return value of `Transaction::watch` is a `TrxWatch` which is a future of type `Future<Item=(), Error=FdbError>`. It consists of `FdbFuture`, so the transaction is not aborted as long as `TrxWatch` alives.

We can revert the behavior, but it will break `Database::transact` if `Transaction::watch` returns retryable error. Maybe we should check if `fdb_transaction_watch` can return retryble error.

Reply:
My biggest concern is that the transaction is canceled in the drop, as that seems to guard against any dangling transactions. I think I'm understanding what your concern is, and you are in a better position to make the call here. But if I have a preference, it's that we try not to have an interface where cancel needs to be called explicitly.

Reply:
I understood and agree. I reverted this change.